Building against new libyui tree

anaselli opened the pull request:
This request also includes:

- removed the needs of legacy buildtools (that means it could be removed if no one uses it, libyui-mga moved too)
- removed all old ycp things
- added new icon loading (from theme or from absolute path) either for GtkImage or GdkPixbuf

What is not managed:

- spec files, since i don't use them. If needed I can work on them on suggestions

shundhammer commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+#
+# Support for pkg-config:
+#
+# Generate a libyui.pc file from libyui.pc.in and install it to
```
NP: Copy&paste leftover (should be libyui-gtk)

anaselli replied:
changed, thanks

shundhammer commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+# flags (-lyui-gtk), compiler flags (including include directories) and paths.
+# See man pkg-config.
+#
+# Notice that libyui-mga linking is needed for libyui-mga-* plugins 
```
NP: Copy&paste leftover: Should be libyui-gtk

anaselli replied:
changed, thanks

shundhammer commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+#pkg_get_variable(YUI_SO_PATCH libyui soversion_patch)
+
+##### This is needed to be set for the libyui core
+SET( SONAME_MAJOR ${YUI_SO_MAJOR} )
```
Is this necessary for this subpackage?

anaselli replied:
Well imo, no. But it has been added to libyui-ncurses.pc and libyui-qt.pc so i changed it a bit to make it looking the same

shundhammer commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+
+set( PKGCONFIG_INSTALL_DIR ${DESTDIR}${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig )
+
+# Generate libyui.pc where some CMake variables are expanded from libyui.pc.in,
```
Copy&paste error again; here it's really confusing because this deals with **both** libyui-gtk and libyui

anaselli replied:
should be fixed that too. Thanks

shundhammer commented on src/CMakeLists.txt:
```diff
@@ -116,26 +123,25 @@ add_library( ${TARGETLIB} SHARED
+##-   )
+##-
+##- add_dependencies( ${TARGETLIB} local-include-dir )
+##- target_include_directories( ${TARGETLIB} BEFORE PUBLIC ${LOCAL_INCLUDE_DIR} )
```
This block is now dead code. Please remove.

anaselli replied:
removed

shundhammer commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+FIND_PACKAGE(PkgConfig REQUIRED)
+
+PKG_CHECK_MODULES(YUI REQUIRED libyui)
+pkg_get_variable(YUI_SO_VERSION libyui soversion)
```
Is any of those `YUI_SO_VERSION` etc.  needed? I don't see it used in libyui-gtk.pc.in. To me it looks as if lines 20-26 can safely be deleted.

anaselli replied:
added them as said above:

```
soversion_major=@SONAME_MAJOR@
soversion_minor=@SONAME_MINOR@
soversion_patch=@SONAME_PATCH@
soversion=@SONAME@
```

shundhammer commented on src/CMakeLists.txt:
```diff
@@ -146,28 +152,24 @@ target_compile_definitions( ${TARGETLIB} PUBLIC VERSION="${VERSION}" )
 
-# Find yui during a combined build
-target_link_directories( ${TARGETLIB} BEFORE PUBLIC ../../libyui/build/src )
+# https://cmake.org/cmake/help/latest/command/link_directories.html suggests to use target_link_libraries
```
This whole block (line 155..161) is not needed for libyui-gtk. We did that for libyui-qt etc. so we can do completely local builds of libyui, libyui-qt, libyui-qt-pkg, libyui-qt-graph, libyui-ncurses, libyui-ncurses-pkg, libyui-rest-api, libyui-rest-api-qt, libyui-rest-api-ncurses without the need of `make install` all the time: This would simply prefer freshly built libraries in ../../libyui/build/...  

But this is not needed or desired here because this subproject (libyui-gtk) is always built separately. It doesn't make any sense to add more link directories (i.e. `-L../../libyui/build/...` parameters) here.

anaselli replied:
First of all, thanks for starting reviewing all. I will take a look of your comments one by one, regarding local deps block i'm evaluating if leaving it because it could be usefull for packagers since it could be easier to add libyui-gtk and libyui-mga plugins as a part of the libyui SRPM (or deb) to build all in one shot and simplify the libyui-bindings building too.

anaselli replied:
Agreed. I added an exception to libyui-mga that could be built into libyui tree so that libyui-bindings can find its includes and lib as well as libyui ones. Will talk about it into another PR (for bindings) i'm opening to start discussion there too

shundhammer replied:
OK.

shundhammer commented:
I also don't know why the Travis build is failing. It might be the docker image; it might have become outdated.

anaselli commented on pkgconfig/CMakeLists.txt:
```diff
@@ -0,0 +1,35 @@
+# CMakeLists.txt for libyui-mga/pkgconfig
```
ok

shundhammer reviewed (approved):
The CMake build environment looks good to me now.

As mentioned in IRC, I can't say anything about the Gtk C++ part; we have to trust your judgement about that. I am sure you built and tested it well enough.

It's unfortunate that the Travis build fails; but I put the blame for that on that docker image used for that. For the libyui core and for all our YaST packages, we migrated to GitHub actions some time ago; that would probably be useful for the community-driven libyui projects as well.

https://github.com/libyui/libyui/tree/master/.github/workflows

But that's independent of this pull request; let's move this forward for now.

anaselli commented:
> As mentioned in IRC, I can't say anything about the Gtk C++ part; we have to trust your judgement about that. I am sure you built and tested it well enough.

I tested some examples and what i use, certainly what i changed :)
 
> It's unfortunate that the Travis build fails; but I put the blame for that on that docker image used for that. For the libyui core and for all our YaST packages, we migrated to GitHub actions some time ago; that would probably be useful for the community-driven libyui projects as well.
> https://github.com/libyui/libyui/tree/master/.github/workflows

In my mind i'd like to fix what is also missing and implemented in Qt, so i could try to understand what kind of error is for what i can of course.
 
> But that's independent of this pull request; let's move this forward for now.
Indeed and i need a starting point first ;)

Thanks

anaselli commented:
I seem i cannot find the changes on master... have I  done some mistakes? I certainly forgot to increase release version but it shouldn't be that i believe...

shundhammer commented:
AFAICS the changes are now in https://github.com/libyui/libyui-gtk/tree/NewLibyuiTree, i.e. in the `NewLibyuiTree` branch of https://github.com/libyui/libyui-gtk.

Another PR from there to the `master` branch of https://github.com/libyui/libyui-gtk should do the trick.

anaselli commented:
https://github.com/libyui/libyui-gtk/pull/79
